instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py

```diff
@@ -265,6 +270,11 @@ def uninstrument_app(app: fastapi.FastAPI):
         app._is_instrumented_by_opentelemetry = False
 
     def instrumentation_dependencies(self) -> Collection[str]:
+        if find_spec("fastapi") is not None:
+            return (_fastapi,)
+        if find_spec("fastapi_slim") is not None:
+            return (_fastapi_slim,)
+        # If neither is installed, return both as potential dependencies
```

Comment:
Do you really need this?

Reply:
you are right, without this, the issue could be solved as well.

Reply:
Ah right, dependencies are OR'ed with the opentelemetry-bootstrap but are AND'ed when instrumenting :sweat: